### ISSUE TITLE
Suggestion for clippy::try_err doesn't account for conversions

### ISSUE THREAD

**Issue by jasoncarr0** [rust-lang/rust-clippy#6994]
Lint name:
clippy::try_err

I tried this code:

```rust
pub fn foo() -> Result<(), Box<dyn Error>> {
  ...
  Err(e)? // e has a concrete error type
  ...
}
```

I expected to see this happen: no lint or clippy would suggest a valid replacement like `return Err(e.into())`

Instead, this happened: Clippy suggests `return Err(e)` which causes a mismatched types error as it doesn't have the casting that try syntax has

### Meta

- `cargo clippy -V`: `clippy 0.0.212 (1c389ff 2020-11-24)`
- `rustc -Vv`:
  ```
rustc 1.50.0-nightly (1c389ffef 2020-11-24)
binary: rustc
commit-hash: 1c389ffeff814726dec325f0f2b0c99107df2673
commit-date: 2020-11-24
host: x86_64-unknown-linux-gnu
release: 1.50.0-nightly
  ```

**Comment by giraffate** [comment]
Can you provide a reproduction on the playground? I guess we've already use `return Err(x)` and `return Err(x.into())` properly, so it would be helpful if there is more information.

**Comment by jasoncarr0** [comment]
I believe this was a mistake on my part, can't reproduce currently
